test/controllers/income/estimatedPay/update/IncomeUpdateCalculatorControllerSpec.scala

```diff
@@ -1334,31 +1439,27 @@ class IncomeUpdateCalculatorControllerSpec
 
   "submitIncomeIrregularHours" must {
     "respond with INTERNAL_SERVER_ERROR for failed request to cache" in {
-      val testController = createTestIncomeUpdateCalculatorController
+
+      val harness = UnifiedHarness.setup()
 
       when(
         journeyCacheService.mandatoryValues(any())(any())
       ).thenReturn(
         Future.failed(new Exception)
       )
 
-      val result: Future[Result] = testController.submitIncomeIrregularHours(1)(
-        RequestBuilder.buildFakeRequestWithOnlySession("GET")
-      )
+      val result: Future[Result] = harness.submitIncomeIrregularHours(1,
+        RequestBuilder.buildFakeRequestWithAuth("GET"))
 
       status(result) mustBe INTERNAL_SERVER_ERROR
     }
 
     "sends Ok on successful submit" in {
-      val testController = createTestIncomeUpdateCalculatorController
 
-      val newAmount = 123
+      when(journeyCacheService.mandatoryValues(Matchers.anyVararg[String])(any()))
```

Comment:
You have encapsulated most of the mock setups to the harness, I would try and do that for every mock that needs to be setup